# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1311113796 -7200
#      Wed Jul 20 00:16:36 2011 +0200
# Node ID 16b33b7ef1b3385fa6c79dd99895ac625752e7b0
# Parent  17bb8974577bf2574d628fd1db2e1fa60b2d1d43
Raised update interval to 500ms.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -238,17 +238,17 @@ MainWindow::construct ()
   connect (settingsAction, SIGNAL (triggered ()), this,
 	   SLOT (processSettingsDialogRequest ()));
   connect (exitAction, SIGNAL (triggered ()), this, SLOT (close ()));
   connect (alignWindowsAction, SIGNAL (triggered ()), this,
 	   SLOT (alignMdiWindows ()));
   connect (reportBugAction, SIGNAL (triggered ()), this,
 	   SLOT (openBugTrackerPage ()));
 
-  setWindowTitle (QString ("Octave GUI (0.0.1)"));
+  setWindowTitle (QString ("Octave GUI (0.0.3)"));
 
   setCentralWidget (m_centralMdiArea);
   addDockWidget (Qt::LeftDockWidgetArea, m_variablesDockWidget);
   addDockWidget (Qt::LeftDockWidgetArea, m_historyDockWidget);
   addDockWidget (Qt::RightDockWidgetArea, m_filesDockWidget);
   setStatusBar (m_statusBar);
 
   readSettings ();
diff --git a/gui/src/MainWindow.h b/gui/src/MainWindow.h
--- a/gui/src/MainWindow.h
+++ b/gui/src/MainWindow.h
@@ -141,27 +141,26 @@ Q_OBJECT public:
   {
   }
 
 protected:
   void run ()
   {
     while (m_mainWindow->isRunning ())
       {
-
 	// Get a full variable list.
 	QList < SymbolRecord > symbolTable =
-	  OctaveLink::instance ()->currentSymbolTable ();
+          OctaveLink::instance ()->copyCurrentSymbolTable ();
 	if (symbolTable.size ())
 	  {
 	    m_mainWindow->variablesDockWidget ()->
 	      setVariablesList (symbolTable);
 	  }
 
         OctaveLink::instance ()->updateHistoryModel ();
-	usleep (100000);
+        usleep (500000);
       }
   }
 private:
   MainWindow * m_mainWindow;
 };
 
 #endif // MAINWINDOW_H
diff --git a/gui/src/OctaveLink.cpp b/gui/src/OctaveLink.cpp
--- a/gui/src/OctaveLink.cpp
+++ b/gui/src/OctaveLink.cpp
@@ -112,17 +112,17 @@ OctaveLink::fetchSymbolTable ()
   std::list < SymbolRecord >::iterator iterator;
   for (iterator = allVariables.begin (); iterator != allVariables.end ();
        iterator++)
     m_symbolTableBuffer.append (iterator->dup ());
   m_symbolTableSemaphore->release ();
   emit symbolTableChanged ();
 }
 
-QList < SymbolRecord > OctaveLink::currentSymbolTable ()
+QList < SymbolRecord > OctaveLink::copyCurrentSymbolTable ()
 {
   QList < SymbolRecord > m_symbolTableCopy;
 
   // Generate a deep copy of the current symbol table.
   m_symbolTableSemaphore->acquire ();
   foreach (SymbolRecord symbolRecord, m_symbolTableBuffer)
     m_symbolTableCopy.append (symbolRecord.dup ());
   m_symbolTableSemaphore->release ();
diff --git a/gui/src/OctaveLink.h b/gui/src/OctaveLink.h
--- a/gui/src/OctaveLink.h
+++ b/gui/src/OctaveLink.h
@@ -101,17 +101,17 @@ public:
   readlineEventHook (void);
   static QString
   octaveValueAsQString (OctaveValue octaveValue);
 
   /**
     * Returns a copy of the current symbol table buffer.
     * \return Copy of the current symbol table buffer.
     */
-  QList < SymbolRecord > currentSymbolTable ();
+  QList < SymbolRecord > copyCurrentSymbolTable ();
 
   void
   updateHistoryModel ();
 
   QStringListModel *
   historyModel ();
 
   void
